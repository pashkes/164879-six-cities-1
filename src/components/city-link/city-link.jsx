import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {getCurrentCity} from "./../../reducer/data/selectors";

export class CityLink extends PureComponent {
  constructor(props) {
    super(props);

    this._handleClickCity = this._handleClickCity.bind(this);
  }

  _handleClickCity(evt) {
    const {onChangeCity, city} = this.props;

    evt.preventDefault();
    onChangeCity(city);
  }

  render() {
    const {city, currentCity} = this.props;

    return (
      <li className="locations__item">
        <button type="button"
          onClick={this._handleClickCity}
          className={`locations__item-link tabs__item ${currentCity === city ? `tabs__item--active` : ``}`}
        >
          <span>{city}</span>
        </button>
      </li>
    );
  }
}

CityLink.propTypes = {
  city: PropTypes.string.isRequired,
  onChangeCity: PropTypes.func.isRequired,
  currentCity: PropTypes.string
};


const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: getCurrentCity(state),
});

export default connect(mapStateToProps)(CityLink);
