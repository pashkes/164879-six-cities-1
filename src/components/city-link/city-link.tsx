import * as React from "react";
import {connect} from "react-redux";

import {getCurrentCity} from "./../../reducer/data/selectors";

interface Props {
  city: string,
  onChangeCity: (city: string) => void,
  currentCity: string,
}

export class CityLink extends React.PureComponent<Props> {
  handleClickCity = (evt) => {
    const {onChangeCity, city} = this.props;

    evt.preventDefault();
    onChangeCity(city);
  };

  render() {
    const {city, currentCity} = this.props;

    return (
      <li className="locations__item">
        <button type="button"
          onClick={this.handleClickCity}
          className={`locations__item-link tabs__item ${currentCity === city ? `tabs__item--active` : ``}`}
        >
          <span>{city}</span>
        </button>
      </li>
    );
  }
}

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: getCurrentCity(state),
});

export default connect(mapStateToProps)(CityLink);
