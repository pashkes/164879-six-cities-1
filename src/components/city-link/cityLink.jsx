import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {ActionCreators} from "../../reducer";

class CityLink extends PureComponent {
  constructor(props) {
    super(props);
    this._onClick = this._onClick.bind(this);
  }

  _onClick(evt) {
    evt.preventDefault();
    this.props.changeCurrentCity(this.props.city);
  }

  render() {
    const {city, currentCity} = this.props;
    return (
      <li className="locations__item">
        <a
          onClick={this._onClick}
          className={`locations__item-link tabs__item ${currentCity === city ? `tabs__item--active` : ``}`}
          href="#">
          <span>{city}</span>
        </a>
      </li>
    );
  }
}

CityLink.propTypes = {
  city: PropTypes.string.isRequired,
  changeCurrentCity: PropTypes.func.isRequired,
  currentCity: PropTypes.string,
};


const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: state.city
});

const mapDispatchToProps = (dispatch) => ({
  changeCurrentCity: (city) => dispatch(ActionCreators.changeCity(city)),
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CityLink);
