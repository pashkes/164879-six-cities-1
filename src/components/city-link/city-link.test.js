import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {CityLink} from "./city-link.jsx";

configure({adapter: new Adapter()});


describe(`city link renders correctly`, () => {
  it(` if link is active`, () => {
    const currentCity = shallow(<CityLink city={`Amsterdam`} currentCity={`Amsterdam`} changeCity={jest.fn} changeCurrentCity={jest.fn} />);
    expect(currentCity).toMatchSnapshot();
  });

  it(` if link isn't active`, () => {
    const defaultCity = shallow(<CityLink city={`Amsterdam`} currentCity={`Paris`} changeCity={jest.fn} />);
    expect(defaultCity).toMatchSnapshot();
  });
});

