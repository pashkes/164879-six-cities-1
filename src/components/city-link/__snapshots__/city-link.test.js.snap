// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`city link renders correctly  if link is active 1`] = `
<li
  className="locations__item"
>
  <button
    className="locations__item-link tabs__item tabs__item--active"
    onClick={[Function]}
    type="button"
  >
    <span>
      Amsterdam
    </span>
  </button>
</li>
`;

exports[`city link renders correctly  if link isn't active 1`] = `
<li
  className="locations__item"
>
  <button
    className="locations__item-link tabs__item "
    onClick={[Function]}
    type="button"
  >
    <span>
      Amsterdam
    </span>
  </button>
</li>
`;
